build: github runner macOS build failed with Cmake 4

found issue in #368 

with github macOS 15 runner image:  '2.328.0', CMake 4.1.1


may relates to https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html

```
- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Clang version >= 16.0 detected, enabling '-Wno-invalid-unevaluated-string'.
-- CMAKE Version: 4.1.1
-- Source Dir: /Users/runner/work/quic/quic/msquic
-- Host System name: Darwin
-- Setting policy 0091
-- System name: Darwin
-- System version: 24.5.0
-- Platform version: 
-- Build type: RelWithDebInfo
-- QUIC Platform: darwin
-- Version Build ID: 0
-- Version Suffix: -private
-- Looking for _SC_PHYS_PAGES
-- Looking for _SC_PHYS_PAGES - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for sysctl
-- Looking for sysctl - found
-- Enabling OpenSsl configuration tests
-- Enabling CA file tests
-- QUIC_ENABLE_LOGGING is false. Disabling logging
-- Disabling (client) shared port support
-- Configuring for OpenSSL 1.1
-- Setting openssldir to /opt/homebrew/etc/openssl@1.1
ERRORWTF darwin 
-- Configuring OpenSSL: /Users/runner/work/quic/quic/msquic/submodules/openssl/config enable-tls1_3;no-makedepend;no-dgram;no-ssl3;no-psk;no-srp;no-zlib;no-egd;no-idea;no-rc5;no-rc4;no-afalgeng;no-comp;no-cms;no-ct;no-srp;no-srtp;no-ts;no-gost;no-dso;no-ec2m;no-tls1;no-tls1_1;no-tls1_2;no-dtls;no-dtls1;no-dtls1_2;no-ssl;no-ssl3-method;no-tls1-method;no-tls1_1-method;no-tls1_2-method;no-dtls1-method;no-dtls1_2-method;no-siphash;no-whirlpool;no-aria;no-bf;no-blake2;no-sm2;no-sm3;no-sm4;no-camellia;no-cast;no-md4;no-mdc2;no-ocb;no-rc2;no-rmd160;no-scrypt;no-seed;no-weak-ssl-ciphers;no-shared;no-tests;--openssldir="/opt/homebrew/etc/openssl@1.1";--prefix=/Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl;-isysroot
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Configuring for OpenSSL
-- Configuring done (6.5s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/runner/work/quic/quic/c_build
ninja: Entering directory `c_build'
[1/86] Building C object msquic/src/core/CMakeFiles/core.dir/ack_tracker.c.o
[2/86] Building C object msquic/src/core/CMakeFiles/core.dir/api.c.o
[3/86] Building C object msquic/src/core/CMakeFiles/core.dir/configuration.c.o
[4/86] Building C object msquic/src/core/CMakeFiles/core.dir/congestion_control.c.o
[5/86] Building C object msquic/src/core/CMakeFiles/core.dir/binding.c.o
[6/86] Building C object msquic/src/core/CMakeFiles/core.dir/crypto.c.o
[7/86] Building C object msquic/src/core/CMakeFiles/core.dir/crypto_tls.c.o
[8/86] Building C object msquic/src/core/CMakeFiles/core.dir/cubic.c.o
[9/86] Building C object msquic/src/core/CMakeFiles/core.dir/bbr.c.o
[10/86] Building C object msquic/src/core/CMakeFiles/core.dir/connection.c.o
[11/86] Building C object msquic/src/core/CMakeFiles/core.dir/datagram.c.o
[12/86] Building C object msquic/src/core/CMakeFiles/core.dir/frame.c.o
[13/86] Building C object msquic/src/core/CMakeFiles/core.dir/library.c.o
[14/86] Building C object msquic/src/core/CMakeFiles/core.dir/listener.c.o
[15/86] Building C object msquic/src/core/CMakeFiles/core.dir/lookup.c.o
[16/86] Building C object msquic/src/core/CMakeFiles/core.dir/loss_detection.c.o
[17/86] Building C object msquic/src/core/CMakeFiles/core.dir/mtu_discovery.c.o
[18/86] Building C object msquic/src/core/CMakeFiles/core.dir/operation.c.o
[19/86] Building C object msquic/src/core/CMakeFiles/core.dir/packet.c.o
[20/86] Building C object msquic/src/core/CMakeFiles/core.dir/packet_space.c.o
[21/86] Building C object msquic/src/core/CMakeFiles/core.dir/packet_builder.c.o
[22/86] Building C object msquic/src/core/CMakeFiles/core.dir/path.c.o
[23/86] Building C object msquic/src/core/CMakeFiles/core.dir/range.c.o
[24/86] Building C object msquic/src/core/CMakeFiles/core.dir/recv_buffer.c.o
[25/86] Building C object msquic/src/core/CMakeFiles/core.dir/registration.c.o
[26/86] Building C object msquic/src/core/CMakeFiles/core.dir/send_buffer.c.o
[27/86] Building C object msquic/src/core/CMakeFiles/core.dir/sent_packet_metadata.c.o
[28/86] Building C object msquic/src/core/CMakeFiles/core.dir/send.c.o
[29/86] Building C object msquic/src/core/CMakeFiles/core.dir/stream.c.o
[30/86] Building C object msquic/src/core/CMakeFiles/core.dir/settings.c.o
[31/86] Building C object msquic/src/core/CMakeFiles/core.dir/stream_recv.c.o
[32/86] OpenSSL configure
Operating system: arm64-apple-darwinDarwin Kernel Version 24.5.0: Tue Apr 22 19:54:15 PDT 2025; root:xnu-11417.121.6~2/RELEASE_ARM64_VMAPPLE
Configuring OpenSSL version 1.1.1w+quic (0x1010117fL) for darwin64-arm64-cc
Using os-specific seed configuration
Creating configdata.pm
Creating Makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL file first)         ***
***                                                                ***
**********************************************************************
[33/86] Building C object msquic/src/core/CMakeFiles/core.dir/stream_send.c.o
[34/86] Building C object msquic/src/core/CMakeFiles/core.dir/stream_set.c.o
[35/86] Building C object msquic/src/core/CMakeFiles/core.dir/timer_wheel.c.o
[36/86] OpenSSL build
FAILED: [code=2] _deps/opensslquic-build/openssl/lib/libssl.a _deps/opensslquic-build/openssl/lib/libcrypto.a /Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl/lib/libssl.a /Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl/lib/libcrypto.a 
cd /Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/submodules/openssl && make install_dev -j3
/opt/homebrew/Cellar/perl/5.40.2/bin/perl "-I." -Mconfigdata "../../../../../msquic/submodules/openssl/util/dofile.pl" \
	    "-oMakefile" ../../../../../msquic/submodules/openssl/include/crypto/bn_conf.h.in > include/crypto/bn_conf.h
/opt/homebrew/Cellar/perl/5.40.2/bin/perl "-I." -Mconfigdata "../../../../../msquic/submodules/openssl/util/dofile.pl" \
	    "-oMakefile" ../../../../../msquic/submodules/openssl/include/crypto/dso_conf.h.in > include/crypto/dso_conf.h
/opt/homebrew/Cellar/perl/5.40.2/bin/perl "-I." -Mconfigdata "../../../../../msquic/submodules/openssl/util/dofile.pl" \
	    "-oMakefile" ../../../../../msquic/submodules/openssl/include/openssl/opensslconf.h.in > include/openssl/opensslconf.h
/Applications/Xcode_16.4.app/Contents/Developer/usr/bin/make depend && /Applications/Xcode_16.4.app/Contents/Developer/usr/bin/make _build_libs
cc  -I. -Iinclude -I../../../../../msquic/submodules/openssl -I../../../../../msquic/submodules/openssl/include -fPIC -arch arm64 -O3 -Wall -isysroot -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_BN_ASM_MONT -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DVPAES_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/opt/homebrew/etc/openssl@1.1\"" -DENGINESDIR="\"/Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl/lib/engines-81.1.1\"" -D_REENTRANT -DNDEBUG   -c -o apps/app_rand.o ../../../../../msquic/submodules/openssl/apps/app_rand.c
cc  -I. -Iinclude -I../../../../../msquic/submodules/openssl -I../../../../../msquic/submodules/openssl/include -fPIC -arch arm64 -O3 -Wall -isysroot -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_BN_ASM_MONT -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DVPAES_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/opt/homebrew/etc/openssl@1.1\"" -DENGINESDIR="\"/Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl/lib/engines-81.1.1\"" -D_REENTRANT -DNDEBUG   -c -o apps/apps.o ../../../../../msquic/submodules/openssl/apps/apps.c
cc  -I. -Iinclude -I../../../../../msquic/submodules/openssl -I../../../../../msquic/submodules/openssl/include -fPIC -arch arm64 -O3 -Wall -isysroot -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_BN_ASM_MONT -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DVPAES_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/opt/homebrew/etc/openssl@1.1\"" -DENGINESDIR="\"/Users/runner/work/quic/quic/c_build/_deps/opensslquic-build/openssl/lib/engines-81.1.1\"" -D_REENTRANT -DNDEBUG   -c -o apps/bf_prefix.o ../../../../../msquic/submodules/openssl/apps/bf_prefix.c
clang: warning: no such sysroot directory: '-DL_ENDIAN' [-Wmissing-sysroot]
clang: warning: no such sysroot directory: '-DL_ENDIAN' [-Wmissing-sysroot]
clang: warning: no such sysroot directory: '-DL_ENDIAN' [-Wmissing-sysroot]
../../../../../msquic/submodules/openssl/apps/bf_prefix.c:10:10: fatal error: 'stdio.h' file not found
   10 | #include <stdio.h>
      |          ^~~~~~~~~
In file included from ../../../../../msquic/submodules/openssl/apps/app_rand.c:10:
In file included from ../../../../../msquic/submodules/openssl/apps/apps.h:13:
In file included from ../../../../../msquic/submodules/openssl/e_os.h:16:
In file included from ../../../../../msquic/submodules/openssl/include/openssl/e_os2.h:243:
/Applications/Xcode_16.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/inttypes.h:24:15: fatal error: 'inttypes.h' file not found
   24 | #include_next <inttypes.h>
      |               ^~~~~~~~~~~~
../../../../../msquic/submodules/openssl/apps/apps.c:18:10: fatal error: 'stdio.h' file not found
   18 | #include <stdio.h>
      |          ^~~~~~~~~
1 error generated.
1 error generated.
make[3]: *** [apps/app_rand.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [apps/bf_prefix.o] Error 1
1 error generated.
make[3]: *** [apps/apps.o] Error 1
make[2]: *** [build_libs] Error 2
[37/86] Building C object msquic/src/core/CMakeFiles/core.dir/version_neg.c.o
[38/86] Building C object msquic/src/core/CMakeFiles/core.dir/worker.c.o
ninja: build stopped: subcommand failed.
make[1]: *** [build-nif] Error 2
===> Hook for compile failed!
```

--- a/build.sh
+++ b/build.sh
@@ -19,6 +19,7 @@ build() {
     fi
     if [ "$(uname -s)" = 'Darwin' ]; then
         JOBS="$(sysctl -n hw.ncpu)"
+        export SDKROOT="$(xcrun --sdk macosx --show-sdk-path)"
     else
         JOBS="$(nproc)"
     fi

